data/transactions/logic/eval.go

```diff
@@ -4522,6 +4548,18 @@ func (cx *EvalContext) stackIntoTxnField(sv stackValue, fs *txnFieldSpec, txn *t
 		}
 		txn.ClearStateProgram = make([]byte, len(sv.Bytes))
 		copy(txn.ClearStateProgram, sv.Bytes)
+	case ApprovalProgramPages:
+		maxPossible := cx.Proto.MaxAppProgramLen * (1 + cx.Proto.MaxExtraAppProgramPages)
+		txn.ApprovalProgram = append(txn.ApprovalProgram, sv.Bytes...)
```

Comment:
Yes. That's the intended behavior, and tests check it. I originally thought I would force adding "full" pages, but decided there was no compelling reason to force it, and it could be convenient to build up programs as you go in arbitrary chunks.

I'll certainly need to describe it more in the specs, as this is a bit different from how other array fields work. But barring objections, I think it's nice.

Reply:
I can see the argument for app_params_get support. I originally thought I would do it. But currently, app_params_get has no notion of an array field, so there is no natural way to express it.

I discussed this level of support with two projects that intend to use it, and they both care only about the 'txn' family. So it seems worth doing this much now.